File: src/kyu_7/find_sum_of_top_left_to_bottom_right_diagonals/java/Diagonal.java
// https://www.codewars.com/kata/5497a3c181dd7291ce000700/train/java

package kyu_7.find_sum_of_top_left_to_bottom_right_diagonals.java;

public class Diagonal {
    public static int diagonalSum(final int[][] matrix) {
        int result = 0;
        for (int i = 0; i < matrix.length; i++) {
            result += matrix[i][i];

        }
        return result;
    }
}

File: src/kyu_6/easy_diagonal/kotlin/Diagonal.kt
// https://www.codewars.com/kata/559b8e46fa060b2c6a0000bf/train/kotlin

package kyu_6.easy_diagonal.kotlin

import java.math.BigInteger

object Diagonal {

    fun diagonal(n: Int, p: Int): BigInteger {
        var result = BigInteger.valueOf(1L)
        var i = BigInteger.valueOf(1L)
        while (i <= BigInteger.valueOf((p + 1).toLong())) {
            result = result * (BigInteger.valueOf(n.toLong()) - i + BigInteger.valueOf(2L)) / i
            i += BigInteger.valueOf(1L)
        }
        return result
    }
}